File: app/src/main/java/com/kipucode/data/local/dao/UserProgressDao.java
package com.kipucode.data.local.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.kipucode.data.local.model.UserProgress;

@Dao
public interface UserProgressDao {
    @Query("SELECT * FROM user_progress WHERE user_id = :userId")
    UserProgress getUserProgress(int userId);
    @Insert
    void insert(UserProgress userProgress);
}

File: app/src/main/java/com/dam/kipucode/data/dao/UserProgressDao.kt
package com.dam.kipucode.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.jg.kipucode.data.local.model.UserProgress

@Dao
interface UserProgressDao {
    @Query("SELECT * FROM user_progress WHERE user_id = :userId")
    fun getUserProgress(userId: String): UserProgress?

    @Insert
    fun insert(userProgress: UserProgress)
}
